git-transport/src/client/blocking_io/http/reqwest.rs

```diff
@@ -193,18 +199,27 @@ mod remote {
             self.make_request(url, headers, true)
         }
 
-        fn configure(&mut self, _config: &dyn Any) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-            if let Some(config) = _config.downcast_ref::<Config>() {
-                self.config = config.into();
+        fn configure(&mut self, config: &dyn Any) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+            if let Some(config) = config.downcast_ref::<Config>() {
+                self.config = config.clone();
             }
             Ok(())
         }
     }
 
     #[derive(Default, Clone)]
     pub struct Config {
-        config: Option<String>,
-        configure: Arc<dyn FnMut(&mut reqwest::ClientBuilder) + Send + Sync + 'static>,
+        pub config: Option<String>,
+        pub configure: Option<
+            Arc<
+                Mutex<
```

Comment:
I introduced a `Mutex` here to address `Cannot borrow data in an ``Arc`` as mutable`. I'm not sure if there are better or preferred alternatives.